charts/elastic/1.2.1/templates/autoscaling-elastic.yaml

```
@@ -0,0 +1,37 @@
+{{- if .Values.elasticsearch.enabled }}
+---
+apiVersion: autoscaling.k8s.elastic.co/v1alpha1
+kind: ElasticsearchAutoscaler
+metadata:
+  name: {{ include "common.names.shortname" $ }}
+  labels:
+  {{- include "common.labels.labels" $ | nindent 4 }}
+  namespace: {{ include "common.names.namespace" $ }}
+spec:
+  {{- with .Values.elasticsearch.params }}
+  ## The name of the Elasticsearch cluster to be scaled automatically.
+  elasticsearchRef:
+    name: {{ include "common.names.shortname" $ }}
+  ## The autoscaling policies.
+  policies:
+  {{- range $k, $v := .roles }}
+    {{- if $v.autoscaling.enabled }}
+    - name: {{ $k }}
+      roles: {{ $v.roles }}
+      resources:
+        nodeCount:
+          min: {{ $v.count }}
+          max: {{ $v.autoscaling.max_nodes }}
```

Try it if max key must be defined and it may => count
```suggestion
          max: {{ default $v.count $v.autoscaling.max_nodes }}
```

if max key doesn't need to be definied in autoscaling.k8s.elastic.co/v1alpha1 manifest
```suggestion
         {{- if  $v.autoscaling.max_nodes  }}
          max: {{ $v.autoscaling.max_nodes }}
         {{- end }}
```

---

Corrected.